### ISSUE TITLE
train navit nest 3d error when backward

### ISSUE THREAD

**Issue by HaloTrouvaille** [lucidrains/vit-pytorch#332]
Thanks for your awesome work!
When I try to train navit_nest_3d model, an error occured:
Function UnbindBackwardAutogradNestedTensor0 returned an invalid gradient at index 0 - got     1  1024
    1  1024
    1  1024
    1  1024
    1  1024
[ torch.LongTensor{5,2} ] but expected shape compatible with [5, j2, 1024]
It seems like an error in backward with https://github.com/lucidrains/vit-pytorch/blob/main/vit_pytorch/na_vit_nested_tensor_3d.py#L327 Do you have any good suggestions, thank you !

**Comment by lucidrains** [comment]
@HaloTrouvaille hey, could you retry on 2.5?

**Comment by lucidrains** [comment]
also, a script that reproduces the error would help immensely

**Comment by HaloTrouvaille** [comment]
Thank you for your reply.
The script is to add the following backward code to the file na_vit_nested_tensor_3d.py
`criterion = nn.CrossEntropyLoss()`
 `optimizer = optim.Adam(v.parameters(), lr=0.001)`
 `targets = torch.tensor([0, 1, 2, 1, 0]).cuda()`
   `loss = criterion(outputs, targets)`
   `optimizer.zero_grad()  # 清零梯度`
   `loss.backward()        # 反向传播`
   `optimizer.step()       # 更新参数`
For some special reasons， I can't use pytorch2.5 temporarily.
Can you help me see if there is any way to replace unbind to solve this problem, thank you!

**Comment by lucidrains** [comment]
@HaloTrouvaille i'm afraid not, sorry, you'll have to upgrade
